### PR TITLE
Provision for installing submariner downstream unreleased build

### PR DESCRIPTION
**Pull request by shylesh**

### CODE REVIEW

**Comment by shylesh** [comment]
Rebased on latest master

File: conf/ocsci/enable_globalnet.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ENV_DATA:
+  enable_globalnet: true
```
Comment (petr-balogh) [inline conf/ocsci/enable_globalnet.yaml:2]: Please document the new config values in `./conf/README.md`

File: conf/ocsci/submariner_downstream_unreleased.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ENV_DATA:
+  submariner_source: "downstream"
```
Comment (petr-balogh) [inline conf/ocsci/submariner_downstream_unreleased.yaml:2]: same if we don't have them

File: ocs_ci/deployment/acm.py [hunk 3]
```diff
@@ -94,9 +97,27 @@ def deploy_downstream(self):
...
+        """
+        This is a prereq for downstream unreleased submariner
+
+        """
+        icsp_data = templating.load_yaml(constants.ACM_DOWNSTREAM_BREW_ICSP)
+        icsp_data_yaml = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="acm_icsp", delete=False
+        )
+        templating.dump_data_to_temp_yaml(icsp_data, icsp_data_yaml.name)
+        run_cmd(f"oc create -f {icsp_data_yaml.name}", timeout=300)
```
Comment (petr-balogh) [inline ocs_ci/deployment/acm.py:119]: we need to add wait for machine config pool.
https://github.com/red-hat-storage/ocs-ci/blob/7f16a45cb770dfd8c23d09405a4ceb71752e4200/ocs_ci/utility/utils.py#L3819

File: ocs_ci/ocs/acm/acm.py [hunk 4]
```diff
@@ -152,6 +155,46 @@ def install_submariner_ui(self, globalnet=True):
...
+                    config.ENV_DATA["submariner_version"],
+                ]
+            )
+            curl_cmd = f"curl --retry 3 --retry-delay 5 -Ls {submariner_full_url}"
+            jq_cmd1 = (
+                'jq -r \'[.raw_messages[].msg | select(.pipeline.status=="complete") |'
+                "{{nvr: .artifact.nvr, index_image: .pipeline.index_image}}] | .[0]'"
+            )
+
+            jq_cmd2 = "jq -r '.index_image.\"v4.14\"'"
```
Comment (petr-balogh) [inline ocs_ci/ocs/acm/acm.py:176]: we should not hardcode the version but rather take from ocp version from config ,
Maybe we can directly update jq command to get the value from `.index_image.v4.14`

File: ocs_ci/templates/ocs-deployment/multicluster/submariner_downstream_unreleased_catsrc.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: submariner-catalogsource
+  namespace: openshift-marketplace
+spec:
+  icon:
+    base64data: ""
+    mediatype: ""
+  image: brew.registry.redhat.io/rh-osbs/iib:PLACE_HOLDER
```
Comment (petr-balogh) [inline ocs_ci/templates/ocs-deployment/multicluster/submariner_downstream_unreleased_catsrc.yaml:10]: Probably format it as a whole string outside and here you can just have PLACEHOLDER
```suggestion
  image: PLACE_HOLDER
```

File: ocs_ci/ocs/acm/acm.py (hunk 4 above)
Comment (shylesh) [reply ocs_ci/ocs/acm/acm.py]: completely replaced with python code, thanks to @prsurve

File: ocs_ci/templates/ocs-deployment/multicluster/submariner_downstream_unreleased_catsrc.yaml (hunk 5 above)
Comment (shylesh) [reply ocs_ci/templates/ocs-deployment/multicluster/submariner_downstream_unreleased_catsrc.yaml]: done

File: ocs_ci/deployment/acm.py (hunk 3 above)
Comment (shylesh) [reply ocs_ci/deployment/acm.py:123]: done

**Comment by shylesh** [comment]
Rebased on latest master

File: ocs_ci/templates/ocs-deployment/multicluster/submariner_downstream_unreleased_catsrc.yaml (hunk 5 above)
Comment (shylesh) [reply ocs_ci/templates/ocs-deployment/multicluster/submariner_downstream_unreleased_catsrc.yaml]: done

File: ocs_ci/ocs/acm/acm.py (hunk 4 above)
Comment (shylesh) [reply ocs_ci/ocs/acm/acm.py]: done

File: ocs_ci/deployment/acm.py (hunk 3 above)
Comment (shylesh) [reply ocs_ci/deployment/acm.py:123]: done

**Review by petr-balogh — DISMISSED** [review]

**Review by dahorak — APPROVED** [review]

File: ocs_ci/deployment/deployment.py [hunk 6]
```diff
@@ -351,6 +352,43 @@ def do_deploy_ocs(self):
                             "ocs_registry_image", None
                         )
                         ocs_install_verification(ocs_registry_image=ocs_registry_image)
+                # if we have Globalnet enabled in case of submariner with RDR
+                # we need to add a flag to storagecluster
+                if config.ENV_DATA.get("enable_globalnet", None):
```
Comment (petr-balogh) [inline ocs_ci/deployment/deployment.py:357]:
```suggestion
                if config.ENV_DATA.get("enable_globalnet", True):
```

File: conf/ocsci/submariner_downstream_unreleased.yaml [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+  submariner_release_type: "unreleased"
+  submariner_version: "0.16.0"
```
[the remark this answers is not in the document]
Comment (keesturam) [reply conf/ocsci/submariner_downstream_unreleased.yaml:4]: For ODF's y stream releases, we will mostly support one unreleased submariner version until GA. However, after the GA we will have to support future versions. 
For eg., all the while with 4.14, we have been using submariner 0.16.0. 
I hope this helps.

**Comment by shylesh** [comment]
Verified here: https://url.corp.redhat.com/rdrglobalnetverification

**Review by prsurve — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Review by am-agrawa — APPROVED** [review]

**Pull request merged**